src/base/emu-i386/simx86/interp.c

```diff
@@ -2262,13 +2262,46 @@ intop3b:		{ int op = ArOpsFR[D_MO(opc)];
 				}
 				break;
 			case Ofs_BX:	/*3*/	 // CALL long indirect restartable
-			case Ofs_BP: {	/*5*/	 // JMP long indirect restartable
+			case Ofs_BP:	/*5*/	 // JMP long indirect restartable
+				if (Fetch(PC+1) >= 0xc0) {
+					CODE_FLUSH();
+					goto illegal_op;
+				}
+				if (REALADDR()) {
+					dosaddr_t oip = 0;
+					int len = ModRM(opc, PC, mode|NOFLDR);
+					if (REG1==Ofs_BX) {
+					    /* ok, now push old cs:eip */
+					    ocs = TheCPU.cs;
+					    oip = PC + len - LONG_CS;
+					    Gen(L_REG, mode, Ofs_CS);
+					    Gen(O_PUSH, mode);
+					    Gen(O_PUSHI, mode, oip);
+					}
+					Gen(L_LXS1, mode, Ofs_EIP);
+					Gen(L_DI_R1, mode);
+					PC = JumpGen(PC, mode, 0x41, len);
+					if (debug_level('e')>2) {
+					    unsigned short jcs = TheCPU.cs;
+					    dosaddr_t jip = PC - LONG_CS;
+					    if (REG1==Ofs_BX)
+						e_printf("CALL_FAR indirect: ret=%04x:%08x\n\tcalling: %04x:%08x\n",
+							 ocs,oip,jcs,jip);
+					    else
+						e_printf("JMP_FAR indirect: %04x:%08x\n",jcs,jip);
+					}
+#ifdef SKIP_EMU_VBIOS
```

Comment:
Do we need that kind of things?

Reply:
I kept it for posterity but probably not. It's only relevant for executing the real VGA BIOS for console graphics and disabled by default.

I remember running the JIT with console graphics a long time ago and it was fast enough for the VGA BIOS, but that was on a CPU a lot faster than the Pentium MMX (or so) that Alberto was working with when he wrote this.